sdk/core/Azure.Core/src/RequestContext.cs

```diff
@@ -14,8 +16,13 @@ namespace Azure
     /// </summary>
     public class RequestContext
     {
+        private List<int>? _customErrorCodes;
```

Comment:
Could these be int[]? I think we should try to keep RequestContext very lightweight. I suspect ConfigureResponse would be called once per array and it would just make a copy of the array. If it's called twice, you would merge the arrays.